dotCMS/src/main/java/com/dotcms/rest/BundleResource.java

```diff
@@ -625,6 +636,144 @@ public Response deleteAllSuccess(@Context final HttpServletRequest request,
                 "Removing bundles in a separated process, the result of the operation will be notified")).build();
     } // deleteAllSuccess.
 
+
+    @Path("/_download/{bundleId}")
+    @GET
+    @JSONP
+    @NoCache
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public final Response downloadBundle(@Context final HttpServletRequest request,
+            @Context final HttpServletResponse response,
+            @PathParam("bundleId") final String bundleId) {
+
+        final InitDataObject initData =
+                new WebResource.InitBuilder(webResource)
+                        .requiredBackendUser(true)
+                        .requestAndResponse(request, response)
+                        .rejectWhenNoUser(true)
+                        .requiredPortlet("publishing-queue")
+                        .init();
+        try {
+            final Bundle bundle = APILocator.getBundleAPI().getBundleById(bundleId);
+            if (!UtilMethods.isSet(bundle)) {
+                throw new DoesNotExistException("Bundle with ID: " + bundleId + " not found");
+            }
+            final File bundleFile = new File(
+                    ConfigUtils.getBundlePath() + File.separator + bundle.getId() + ".tar.gz");
```

Comment:
Codacy found an issue: [The String literal ".tar.gz" appears 5 times in this file; the first occurrence is on line 664](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6239861)